### PR TITLE
Gisida lite legend

### LINKED ISSUE #182: Add Legend Components to Gisida lite
**Is this a new feature or an enhancement?**
feature
**Is your feature request related to a problem? Please describe.**
The legend component is currently being used in three projects i.e [Akuko](https://github.com/onaio/akuko-app/tree/develop/src/containers/Post/Map/components/Legend), [Grp](https://github.com/onaio/grp-frontend/blob/tab-filter-task/src/containers/submissions/MapWrapper/components/Legend/index.tsx), and [Gisida](https://github.com/onaio/gisida-react/tree/staging/src/lib/components/Legend). Moving these to js tools will allow for re-usability and reduce duplication of efforts.

**Describe the solution you'd like**
Move the various legend types components to Gisidalite.

### PR DESCRIPTION
**Pull request by kahummer**
**closes #182**

Screenshot from 2021-08-04 16-11-28

**Changes included with this PR**
- addition of a legend component, tests, styles, and utilities for the same
- moving legend and map component to respective folders
- export map and legend components

**Checklist**

- [x] tests are included and passing
- [x] documentation is changed or added
- [x] a .storybook story was added where necessary.
- [x] code is [transpiled](https://github.com/onaio/js-tools#transpiling-the-package-and-generating-type-definations-for-it)

### CODE REVIEW

File: packages/GisidaLite/src/legend/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  label: string;
+}
+/** Legend props */
+export interface LegendProps {
+  legendData: LegendBlock[];
+  legendCreditText?: string;
+}
+
+/** Default props for GisidaLite */
+export const defaultLegendProps: LegendProps = {
```
Comment (moshthepitt) [inline packages/GisidaLite/src/legend/index.tsx:18]: How were these chosen?

Comment (kahummer) [reply packages/GisidaLite/src/legend/index.tsx:18]: For grp we are using [ckmeans](https://www.npmjs.com/package/ckmeans) for clustering the data then building the labels based on next and current index. The colors are coming from config and the width is calculated by dividing 100 by colors.length. I think this use case is a bit custom and maybe handling this separately from this package would be ideal

Comment (moshthepitt) [reply packages/GisidaLite/src/legend/index.tsx:18]: How would that look like?

Comment (kahummer) [reply packages/GisidaLite/src/legend/index.tsx:18]: Maybe adding an issue that adds utils to build this based on the various codebases would be necessary. Implementing this in a custom way may require changes in the future which may require changing  the code or  reverting back to overriding the legend data for most use cases. The plan I'd go with is create a discussion ticket around this so we can establish a standard approach with which this package can be used across the various projects.

Comment (kahummer) [reply packages/GisidaLite/src/legend/index.tsx:18]: I've added this issue here https://github.com/onaio/js-tools/issues/186

**Review by moshthepitt — APPROVED** [review]

File: packages/GisidaLite/README.md [hunk 2]
```diff
@@ -27,3 +29,42 @@ interface GisidaLiteProps {
...
    - [layer](https://github.com/alex3165/react-mapbox-gl/blob/master/docs/API.md#layer)
    - [GeoJSONLayer](https://github.com/alex3165/react-mapbox-gl/blob/master/docs/API.md#geojsonlayer)
+
+## Legend
+
+A simple `React` component that takes an object of color, width and label as props then outputs a map legend.
+
+### Usage
+
+To get the legend to work, the following props should be provided
```
Comment (kelvin-muchiri) [inline packages/GisidaLite/README.md:39]: Could we add a description of each prop?

**Review by kelvin-muchiri — APPROVED** [review]

**Pull request merged**